src/plugins/condition/components/TestData.vue

```diff
@@ -75,17 +124,117 @@
 export default {
     inject: ['openmct'],
     props: {
-        isEditing: Boolean
+        isEditing: Boolean,
+        telemetry: {
+            type: Array,
+            required: true,
+            default: () => []
+        },
+        testData: {
+            type: Object,
+            required: true,
+            default: () => {
+                return {
+                    applied: false,
+                    conditionTestInputs: []
+                }
+            }
+        }
     },
     data() {
         return {
             expanded: true,
-            isApplied: true
+            isApplied: false,
+            testInputs: [],
+            telemetryMetadataOptions: {}
         };
     },
+    watch: {
+        isEditing(editing) {
+            if (!editing) {
+                this.resetApplied();
+            }
+        },
+        telemetry: {
+            handler() {
+                this.initializeMetadata();
+            },
+            deep: true
+        },
+        testData: {
+            handler() {
+                this.initialize();
+            },
+            deep: true
+        }
+    },
+    beforeDestroy() {
+        this.resetApplied();
+    },
+    mounted() {
+        this.initialize();
+        this.initializeMetadata();
+    },
     methods: {
-        applyTestData(ev) {
-            this.$emit('change', ev.target.checked);
+        applyTestData() {
+            this.isApplied = !this.isApplied;
+            this.persist();
+        },
+        initialize() {
+            if (this.testData && this.testData.conditionTestInputs) {
+                this.testInputs = this.testData.conditionTestInputs;
+            }
+            if (!this.testInputs.length) {
+                this.addTestInput();
+            }
+        },
+        initializeMetadata() {
+            this.telemetry.forEach(telemetryObj => {
+                this.openmct.objects.get(telemetryObj.identifier).then((telemetryObject) => {
```

Comment:
I don't think we need to fetch these objects again here? We really need to keep object fetches to a minimum, they take time, especially with a slow backend.

Reply:
Removed the fetches. I thought they were identifiers - I'll be sure to check in the future.